### PR TITLE
solution w/o Router

### PR DESCRIPTION
**Pull request by eugenekh81**
[DEMO LINK]
https://eugenekh81.github.io/react_pagination/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,6 +1,67 @@
...
+      setCurrentPage(page);
+    }, [],
+  );
+  return (
+    <div className="App">
+
+      <h1>Pagination</h1>
+
+      <label htmlFor="withInfo">
+        {'Show additional information: '}
```
Comment (ArtemTopolskyi) [inline src/App.tsx:26]:
```suggestion
        Show additional information:
```

**Review by ArtemTopolskyi — CHANGES_REQUESTED** [review]
image
при нажатии на многоточие - пагинация ломается и отображается NaN

File: src/App.tsx [hunk 2]
```diff
@@ -1,6 +1,68 @@
...
+    }, [],
+  );
+  return (
+    <div className="App">
+
+      <h1>Pagination</h1>
+
+      <label htmlFor="withInfo">
+        Show additional information:
+        {' '}
```
Comment (ArtemTopolskyi) [inline src/App.tsx:27]: use template string instead

File: src/App.tsx [hunk 3]
```diff
@@ -1,6 +1,68 @@
...
+          )}
+        >
+          <option value="3">3</option>
+          <option value="5">5</option>
+          <option value="10">10</option>
+          <option value="20">20</option>
+        </select>
+      </label>
+
+      <Pagination
```
Comment (ArtemTopolskyi) [inline src/App.tsx:55]: add Pagination component to pull request
image

**Review by ArtemTopolskyi — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (eugenekh81) [reply src/App.tsx:55]: oops ) added )

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]
if you implement this optional task
> (*) Change view to show buttons only for the first, last, current, one precurrent and postcurrent pages
< 1 ... 4 [5] 6 ... 12 >
< [1] 2 ... 12 >

it doesn't work as intended
Screenshot 2022-06-03 at 13 08 22
Screenshot 2022-06-03 at 13 08 17
Screenshot 2022-06-03 at 13 08 06

**Pull request closed without merging**